pkg/sysfs/parsers.go

```diff
@@ -166,3 +327,76 @@ func ParseFileEntries(path string, values map[string]interface{}, pickFn PickEnt
 
 	return nil
 }
+
+// ParseFileByLines parses a sysfs files for the given entries.
+func ParseFileByLines(path string, splitFn SplitFieldsFn, pickFn PickFieldFn) error {
+	var err error
+
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		sysfsError(path, "failed to read file: %v", err)
+	}
+
+	for _, line := range strings.Split(string(data), "\n") {
+		fields, err := splitFn(line)
+		switch {
+		case err == ErrSkip:
+			continue
+		case err != nil:
+			return nil
+		}
+		// Sort according to caller-provided Index which indicates in which order the caller
+		// wants to consume (usually because the caller statefully combines the fields).
+		sort.Slice(fields, func(i, j int) bool { return fields[i].Index <= fields[j].Index })
+
+		for idx, f := range fields {
+			ptr, err := pickFn(idx, f)
+			switch {
+			case err == ErrSkip:
+				continue
+			case err != nil:
+				return err
+			}
+
+			switch ptr.(type) {
```

Comment:
Would be possible to use the `p := ptr.(type)` construct here, agani?

Reply:
I wonder how long it will take to get this into my thick skull... if ever. ;-)
I'll update the code.